app/src/main/java/com/infomaniak/mail/ui/bottomSheetDialogs/ScheduleSendBottomSheetDialog.kt

```diff
@@ -0,0 +1,224 @@
+        }
+
+        val shouldDisplayDivider = lastScheduleItem.isVisible
+        (scheduleItems.children.first() as ActionItemView).setDividerVisibility(shouldDisplayDivider)
+    }
+
+    companion object {
+        const val MIN_SCHEDULE_DELAY_MINUTES = 5
+        const val MAX_SCHEDULE_DELAY_YEARS = 10
+    }
+}
+
+enum class TimeToDisplay {
+    NIGHT,
+    MORNING,
+    AFTERNOON,
+    EVENING,
+    WEEKEND;
+
+    companion object {
+        fun getTimeToDisplayFromDate(): TimeToDisplay {
+            val now = Date()
+            val timeSlot = Date(now.time)
+            return if (now.isWeekend()) WEEKEND else when (now) {
+                in timeSlot.setHour(0).setMinute(0)..timeSlot.setHour(7).setMinute(54) -> NIGHT
+                in timeSlot.setHour(7).setMinute(55)..timeSlot.setHour(13).setMinute(54) -> MORNING
+                in timeSlot.setHour(13).setMinute(55)..timeSlot.setHour(17).setMinute(54) -> AFTERNOON
+                in timeSlot.setHour(17).setMinute(55)..timeSlot.setHour(23).setMinute(54) -> EVENING
+                else -> NIGHT
+            }
+        }
+    }
+}
+
+enum class Schedule(
+    @StringRes val scheduleTitleRes: Int,
+    @DrawableRes val scheduleIconRes: Int,
+    val date: Date,
+    val timeToDisplay: List<TimeToDisplay>,
+    val matomoValue: String,
+) {
+    LATER_THIS_MORNING(
+        R.string.laterThisMorning,
+        R.drawable.ic_morning_sunrise_schedule,
+        Date().getMorning(),
+        listOf(TimeToDisplay.NIGHT),
+        "laterThisMorning",
+    ),
+    MONDAY_MORNING(
+        R.string.mondayMorning,
+        R.drawable.ic_morning_schedule,
+        Date().getNextMonday().getMorning(),
+        listOf(TimeToDisplay.WEEKEND),
+        "nextMondayMorning",
+    ),
+    MONDAY_AFTERNOON(
+        R.string.mondayAfternoon,
+        R.drawable.ic_afternoon_schedule,
+        Date().getNextMonday().getAfternoon(),
+        listOf(TimeToDisplay.WEEKEND),
+        "nextMondayAfternoon",
+    ),
+    THIS_AFTERNOON(
+        R.string.thisAfternoon,
+        R.drawable.ic_afternoon_schedule,
+        Date().getAfternoon(),
+        listOf(TimeToDisplay.MORNING),
+        "thisAfternoon",
+    ),
+    THIS_EVENING(
+        R.string.thisEvening,
+        R.drawable.ic_evening_schedule,
+        Date().getEvening(),
+        listOf(TimeToDisplay.AFTERNOON),
+        "thisEvening",
+    ),
+    TOMORROW_MORNING(
+        R.string.tomorrowMorning,
+        R.drawable.ic_morning_schedule,
+        Date().tomorrow().getMorning(),
+        listOf(TimeToDisplay.NIGHT, TimeToDisplay.MORNING, TimeToDisplay.AFTERNOON, TimeToDisplay.EVENING),
+        "tomorrowMorning",
+    ),
+    NEXT_MONDAY(
+        R.string.nextMonday,
+        R.drawable.ic_arrow_return,
+        Date().getNextMonday().getMorning(),
+        listOf(TimeToDisplay.NIGHT, TimeToDisplay.MORNING, TimeToDisplay.AFTERNOON, TimeToDisplay.EVENING),
+        "nextMonday",
+    )
```

Comment:
Dates are computed a single time when the enum is instanciated this should be computed dynamically